automated install/basic-install.sh

```diff
@@ -473,7 +473,10 @@ update_repo() {
     git stash --all --quiet &> /dev/null || true # Okay for stash failure
     git clean --quiet --force -d || true # Okay for already clean directory
     # Pull the latest commits
-    git pull --quiet &> /dev/null || return $?
+    if ! git branch | grep --quiet -E '^[*] [(]HEAD detached at pull/[[:digit:]]+/merge[)]$'; then
```

Comment:
Just for the hard of thinking (me!) Can you explain the pattern that is being matched here, please?

Reply:
It is supposed to match `* (HEAD detached at pull/#/merge)` from the start to the end of the line with # being the number (as a free amount of digits, but at least one) of the pull request the test is run for. The `[` and `]` are supposed to create character classes and therefore quote `*`, `(` and `)` in the regex, but there might be a better way to do this.